utils/components.py

```diff
@@ -61,3 +69,311 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
             f"ilab --config=DEFAULT model download --repository {repository} --release {release} --model-dir {base_model.path}"
         ],
     )
+
+
+@dsl.component(base_image=PYTHON_IMAGE)
+def test_judge_teacher_models(cm_name: str, secret_name: str):
+    import base64
+    import json
+    import os
+    import sys
+
+    import requests
+    from kubernetes import client
+    from kubernetes.client.rest import ApiException
+
+    sa_path = "/var/run/secrets/kubernetes.io/serviceaccount"
+    configuration = client.Configuration()
+    configuration.api_key["authorization"] = open(f"{sa_path}/token").readline()
+    configuration.api_key_prefix["authorization"] = "Bearer"
+    configuration.host = f"https://{os.environ['KUBERNETES_SERVICE_HOST']}"
+    configuration.ssl_ca_cert = f"{sa_path}/ca.crt"
+
+    model_endpoint = ""
+    model_name = ""
+    model_api_key = ""
+    namespace = open(f"{sa_path}/namespace").readline()
+
+    with client.ApiClient(configuration) as api_client:
+        core_api = client.CoreV1Api(api_client)
+
+        try:
+            config_map = core_api.read_namespaced_config_map(cm_name, namespace)
+            print(f"Reading configmap {cm_name} data...")
+            model_endpoint = config_map.data["endpoint"]
+            model_name = config_map.data["model"]
+        except ApiException as e:
+            print(f"""
+            ############################################ ERROR ###########################################################
+            # Configmap {cm_name} does not exist. Ensure you created a configmap with this name in namespace {namespace} #
+            ##############################################################################################################
+            """)
+            sys.exit(1)
+
+        try:
+            secret = core_api.read_namespaced_secret(secret_name, namespace)
+            print(f"Reading secret {secret_name} data...")
+            model_api_key = base64.b64decode(secret.data["api_key"]).decode("utf-8")
+        except ApiException as e:
+            print(f"""
+            ############################################## ERROR #######################################################
+            # Secret {secret_name} does not exist. Ensure you created a secret with this name in namespace {namespace} #
+            ############################################################################################################
+            """)
+            sys.exit(1)
+
+    request_auth = {"Authorization": f"Bearer {model_api_key}"}
+    request_body = {
+        "model": model_name,
+        "messages": [{"role": "user", "content": "tell me a funny joke."}],
+    }
+    resp = requests.post(
```

Comment:
Optional: It'd be nice to at least try this a few times before failing in case of a network flake.